src/m3ninx/index/segment/fst/segment.go

```diff
@@ -753,7 +753,7 @@ func (r *fsSegment) retrieveTermsBytesWithRLock(base []byte, offset uint64) (pro
 	const sizeofUint64 = 8
 	var (
 		magicNumberEnd   = int64(offset) // to prevent underflows
-		magicNumberStart = offset - sizeofUint64
+		magicNumberStart = magicNumberEnd - sizeofUint64
```

Comment:
Do I understand it correctly that this change only prevents the panic, but would still cause an error on a check below (with `magicNumberStart` being negative)? If so, maybe you have an idea how could such situation happen in the first place?
P.S. I'm not asking to spend a lot of time investigating it and would take a honest "no" as an answer here, just wanted to get a better picture. I guess in that case we would eventually get a more descriptive error and avoid a panic, which is also a good improvement.

Reply:
> Do I understand it correctly that this change only prevents the panic, but would still cause an error on a check below (with magicNumberStart being negative)?

Yes

> If so, maybe you have an idea how could such situation happen in the first place?

I guess it might be due to some race condition, since it's happening on Buildkite, but not locally. Though I can only speculate, since I'm unfamiliar with this part of the code.